packages/virtualized-lists/Lists/VirtualizedList.js

```diff
@@ -276,10 +276,23 @@ class VirtualizedList extends StateSafePureComponent<Props, State> {
 
     scrollRef.scrollTo({
       animated,
-      ...this._cartesianScrollOffset(offset),
+      ...this._scrollToParamsFromOffset(offset),
     });
   }
 
+  _scrollToParamsFromOffset(offset: number): {x?: number, y?: number} {
+    const {horizontal, rtl} = this._orientation();
+    if (horizontal && rtl) {
```

Comment:
This change broke lists in our iOS app in RTL. We heavily rely on horizontal lists and ship to both iOS and Android in LTR and RTL. This line should've been `if (horizontal && rtl && Platform.OS !== 'ios') {`. We tested that fix in our app and that has resolved things for us. @NickGerleman is Meta using large horizontal lists in any part of its localized RN surfaces? If so, is there something I'm missing here?

Reply:
Able to repro native `scrollTo` not using right coordinate space on iOS old arch. New arch has different scrollview component, and seems to be doing the right thing.

There are a good number of changes since this one, but there was an effort to get all the events/platforms onto consistent coordinate spaces. So, no "if iOS, on Paper, do opposite".

I can make quick PR to try to fix this.

Reply:
Try out this change: https://github.com/facebook/react-native/pull/42094/files

Reply:
Can confirm that change also fixes it